File: 백준/Bronze/1978. 소수 찾기/소수 찾기.js
const { log } = require("console");
let fs = require("fs");
const { exit } = require("process");
let input = fs.readFileSync("/dev/stdin").toString().split("\n");

const isPrime = (n) => {
  if (n === 1) {
    return false;
  }

  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
};

let n = Number(input[0]);
let arr = input[1].split(" ").map((x) => Number(x));

console.log(arr.filter((v) => isPrime(v)).length);

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        sc.nextLine();

        String[] numbers = sc.nextLine().split(" ");
        int count = 0;

        for(int i = 0; i<n; i++) {
            int number = Integer.parseInt(numbers[i]);
            if(isPrime(number)) {
                count++;
            }
        }

        System.out.println(count);

        sc.close();
    }

    public static boolean isPrime(int number) {
        if(number < 2) {
            return false;
        }
        for(int i = 2; i<=Math.sqrt(number); i++) {
            if(number % i == 0) {
                return false;
            }
        }
        return true;
    }
}
